cli/src/main.cpp

```diff
@@ -42,6 +47,8 @@ int main(int argc, char *argv[])
 	const QCommandLineOption ignoreErrorOption(QStringList() << "ignore-error", "don't exit on error.");
 	const QCommandLineOption proxyOption(QStringList() << "proxy", "Use given proxy.", "[user:password]@host:port", "");
 	const QCommandLineOption noLoginOption(QStringList() << "no-login", "disable auto login.");
+	const QCommandLineOption jsonOption(QStringList() << "j" << "json", "output results as json.");
```

Comment:
This only only works for images loading, right?

So if I use for example `--return-tags --json`, it will still load images? I don't really mind, but maybe this could be made clear in the description/option name, as some users might try to do it.

Reply:
What about returning "not-implemented" (with quotes) instead?

Reply:
That's a solution.

But given I'm in the process of implementing JSON serialization to most classes, I will simply add the support to the other flags after merging the review 👍